src/midi/midi_fluidsynth.cpp

```diff
@@ -368,19 +373,20 @@ void MidiHandlerFluidsynth::PrintStats()
 	soft_limiter.PrintStats();
 }
 
-void MidiHandlerFluidsynth::MixerCallBack(uint16_t frames)
+void MidiHandlerFluidsynth::MixerCallBack(uint16_t requested_frames)
 {
-	constexpr uint16_t max_samples = expected_max_frames * 2; // two channels per frame
-	std::array<float, max_samples> stream;
-
-	while (frames > 0) {
-		constexpr uint16_t max_frames = expected_max_frames; // local copy fixes link error
-		const uint16_t len = std::min(frames, max_frames);
-		fluid_synth_write_float(synth.get(), len, stream.data(), 0, 2,
-		                        stream.data(), 1, 2);
-		const auto &out_stream = soft_limiter.Apply(stream, len);
-		channel->AddSamples_s16(len, out_stream.data());
-		frames -= len;
+	constexpr uint16_t max_frames = expected_max_frames; // local fixes link error
+	std::vector<float> render_buffer(max_frames * 2);    // L & R channels
+
+	while (requested_frames > 0) {
+		const uint16_t frames = std::min(requested_frames, max_frames);
+		assert(frames <= max_frames);
```

Comment:
I think this assert is an overkill :) It's calculated literally a line above…

Reply:
Agreed; knocking it out :-)